packages/frontend-api/src/Model/Product/Filter/ProductFilterDataMapper.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrontendApiBundle\Model\Product\Filter;
+
+use Shopsys\FrameworkBundle\Model\Product\Brand\BrandFacade;
+use Shopsys\FrameworkBundle\Model\Product\Filter\ParameterFilterData;
+use Shopsys\FrameworkBundle\Model\Product\Filter\ProductFilterData;
+use Shopsys\FrameworkBundle\Model\Product\Flag\FlagFacade;
+use Shopsys\FrameworkBundle\Model\Product\Parameter\ParameterFacade;
+
+class ProductFilterDataMapper
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Product\Flag\FlagFacade
+     */
+    protected FlagFacade $flagFacade;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Product\Brand\BrandFacade
+     */
+    protected BrandFacade $brandFacade;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Product\Parameter\ParameterFacade
+     */
+    protected ParameterFacade $parameterFacade;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Product\Parameter\Parameter[]
+     */
+    protected array $parametersByUuid = [];
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Product\Parameter\ParameterValue[]
+     */
+    protected array $parameterValuesByUuid = [];
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Flag\FlagFacade $flagFacade
+     * @param \Shopsys\FrameworkBundle\Model\Product\Brand\BrandFacade $brandFacade
+     * @param \Shopsys\FrameworkBundle\Model\Product\Parameter\ParameterFacade $parameterFacade
+     */
+    public function __construct(
+        FlagFacade $flagFacade,
+        BrandFacade $brandFacade,
+        ParameterFacade $parameterFacade
+    ) {
+        $this->flagFacade = $flagFacade;
+        $this->brandFacade = $brandFacade;
+        $this->parameterFacade = $parameterFacade;
+    }
+
+    /**
+     * @param array $frontendApiFilter
+     * @return \Shopsys\FrameworkBundle\Model\Product\Filter\ProductFilterData
+     */
+    public function mapFrontendApiFilterToProductFilterData(array $frontendApiFilter): ProductFilterData
+    {
+        $productFilterData = new ProductFilterData();
+        $productFilterData->minimalPrice = $frontendApiFilter['minimalPrice'] ?? null;
+        $productFilterData->maximalPrice = $frontendApiFilter['maximalPrice'] ?? null;
+        $productFilterData->brands = isset($frontendApiFilter['brands']) ? $this->brandFacade->getByUuids(
+            $frontendApiFilter['brands']
+        ) : [];
```

Comment:
_nitpick_

I'm not sure whether this formatting is the best... how about
```
$productFilterData->brands = isset($frontendApiFilter['brands'])
    ? $this->brandFacade->getByUuids($frontendApiFilter['brands'])
    : [];
```

Reply:
This formatting has been chosen by one of our fixers.
d74056991f30f396850621a300b95f3031bd9fae

Reply:
rather use if/else statement which is more readable in this case, please

Reply:
ok, I don't insist on change. 
if/else seems more appropriate in this case. 
But I'm not happy about the overall formatting of long ternary operators... but that's a discussion for another time

Reply:
I would not allow to use multiline ternary operators. It should be used just for really obvious decisions. In cases like this it goes against readability.

We definetly should kick it off some day.

Reply:
fb3db3a5bd6d319db7a37f990b4a5766a0140dc6